corn-maze.lic

```diff
@@ -0,0 +1,632 @@
+        while thing = DRC.left_hand || DRC.right_hand
+          if @looting
+            case thing
+            when *@junk_list
+              drop_item(thing)
+            else
+              stow_thing(thing)
+            end
+          elsif DRC.left_hand.include?('pass') || DRC.right_hand.include?('pass')
+            stow_thing(thing)
+          else
+            drop_item(thing)
+          end
+        end
+      end
+    end
+    @last_looted = XMLData.room_count
+  end
+
+  def drop_item(item)
+    case bput("drop my #{item}", 'You drop', 'would damage it', 'smashing it to bits', 'Something appears different about')
+    when 'would damage it', 'Something appears different about'
+      drop_item(item)
+    end
+  end
+
+  def wait_on_done_move
+    last_room = XMLData.room_count
+    pause 0.5
+    while last_room != XMLData.room_count
+      last_room = XMLData.room_count
+      pause 0.5
+    end
+  end
+
+  def get_task
+    wait_on_done_move
+    @halfling_path.each do |x|
+      exit if Flags['maze_done']
+      if DRRoom.npcs.include? 'Halfling'
+        break
+      else
+        custom_move "#{x}"
+      end
+    end
+    fput 'ask halfling about task'
+    echo "Task count = #{@task_count}"
+    if @searching == false
+      case bput('ask halfling for task', 'A cheerful Halfling smiles happily', 'To whom are you', 'A cheerful Halfling looks at you and says')
+      when 'A cheerful Halfling looks at you and says'
+        fput('ask halfling for task cancel')
+        fput('ask halfling for task cancel')
+        return get_task
+      end
+    end
+    @done = false
+    find_start
+    @done = false
+    wait_for_command
+  end
+
+  def find_start
+    wait_on_done_move
+    unless XMLData.room_title.include?('Labyrinth') || XMLData.room_title.include?('Corn Maze, Entrance')
+      DRC.message("The script can't find its way back from here!  Make your way back towards the northwest Labyrinth using elanthipedia's Corn Maze entry, such as https://elanthipedia.play.net/Sleeping_Dragon_Corn_Maze_432 and then unpause the script!")
+      pause_script('cornmaze')
+    end
+    until @done
+      exit if Flags['maze_done']
+      find_exits = DRC.bput("look",/Obvious paths: .*\./,)
+      exits = find_exits.match(/^Obvious paths: (.*)\./).to_a[1].split(', ')
+      if (exits.include? 'north')
+        custom_move 'north'
+      elsif (exits.include? 'east')
+        custom_move 'east'
+      elsif (exits.include? 'south')
+        custom_move 'south'
+        custom_move 'south'
+        custom_move 'east'
+      elsif exits.include?('west') && !exits.include?('southwest') && !exits.include?('east') && !exits.include?('south') && !exits.include?('north') && !exits.include?('southeast') && !exits.include?('northwest') && !exits.include?('northeast')
```

Comment:
💡 You could use intersection to see if any elements in one array are in another:
```ruby
shared = [1,2,3,'A'] & (1..9).to_a
shared     # => [1, 2, 3]
```

In our case, `(exits & ['west', 'southwest', ...]).empty?`. Do you think that is any clearer?

Or `elsif ['west', 'southwest', ...].none? { |dir| exits.include?(dir) }`
```

Reply:
I'll look into changing how it was done there.